src/R/Editor/Impl/Extensions/StringExtensions.cs

```diff
@@ -16,5 +17,27 @@ public static class StringExtensions {
             }
             return name;
         }
+
+        /// <summary>
+        /// Locates boundaries of '{ }' block in the text.
+        /// Takes into account curly brace nesting.
+        /// </summary>
+        /// <param name="text">Text to look into</param>
+        public static ITextRange GetScopeBlockRange(this string text) {
+            var start = text.IndexOf('{');
+            if (start < 0) {
+                return TextRange.FromBounds(0, 0);
+            }
+
+            var bc = new BraceCounter<char>('{', '}');
+            var end = start;
+            bc.CountBrace(text[end]);
+            while (bc.Count > 0 && end < text.Length - 1) {
+                end++;
+                bc.CountBrace(text[end]);
```

Comment:
Don't you need to re-instantiate `BraceCounter` in the loop?

Reply:
No, it counts braces in the loop (it is a stack effectively) `bc.Count == 0` when braces match.

Reply:
Ok, do you care about cases like `}{}`?

Reply:
Not really. It is up to the caller. The code will detect `{}`